### PR TITLE
[DPM] Entities and caches declaration for DPM's cache

### PR DESCRIPTION
**Pull request by cj-chung**
Declare required entities and caches for DPM's cache.

### CODE REVIEW

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalPorts.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.futurewei.alcor.dataplane.entity;
+
+import java.util.List;
+
+public class InternalPorts {
+    private String subnetId;
+    private String securityGroupId;
+    private List<String> ports;
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalPorts.java:8]: ports => portIds

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalSubnets.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.futurewei.alcor.dataplane.entity;
+
+import java.util.List;
+
+public class InternalSubnets {
+    private String vpcId;
+    private String routerId;
+    private List<String> subnets;
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalSubnets.java:8]: subnets => subnetIds

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/RouterSubnetsCache.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Repository
+@ComponentScan(value="com.futurewei.alcor.common.db")
+public class RouterSubnetsCache implements ICacheRepository<InternalSubnets> {
+    private final ICache<String, InternalSubnets> routerSubnetsCache;
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/RouterSubnetsCache.java:38]: For each cache, can we add a short description/comment to explain what key and value represents?

Comment (xieus) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/RouterSubnetsCache.java:38]: Here the key is router id, right?

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/SecurityGroupPortsCache.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+import java.util.Map;
+
+@Repository
+@ComponentScan(value="com.futurewei.alcor.common.db")
+public class SecurityGroupPortsCache {
+    private ICache<String, InternalPorts> securityGroupPortsCache;
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/SecurityGroupPortsCache.java:32]: Is the key security group id?

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/VpcSubnetsCache.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Repository
+@ComponentScan(value="com.futurewei.alcor.common.db")
+public class VpcSubnetsCache implements ICacheRepository<InternalSubnets> {
+    private final ICache<String, InternalSubnets> vpcSubnetsCache;
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/VpcSubnetsCache.java:38]: Is the key vpc id?

**Review by xieus — COMMENTED** [review]
A few comments.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalPorts.java (hunk 1 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalPorts.java]: fixed

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/RouterSubnetsCache.java (hunk 3 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/RouterSubnetsCache.java]: yes. I have added a short description for each cache declaration.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/SecurityGroupPortsCache.java (hunk 4 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/SecurityGroupPortsCache.java:33]: yes

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/VpcSubnetsCache.java (hunk 5 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/VpcSubnetsCache.java:39]: yes

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalSubnets.java (hunk 2 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/InternalSubnets.java]: fixed

**Review by xieus — APPROVED** [review]
LGTM.

**Pull request merged**